### ISSUE TITLE
Param spec with additional parameters have incorrect suggestion.

### ISSUE THREAD

**Issue by s3rius** [microsoft/pyright#3772]
Hi and thanks for creating such a beautiful tool.

I'm building a package and want to add additional parameters to function. To provide correct type hints I use [PEP-612](https://peps.python.org/pep-0612/). It works in Python, but it breaks suggestions in pylance

## Environment data

-   Language Server version: v2022.7.43 (pre-release)
-   OS and version: ArchLinux 
-   VSCode version: 1.69
-   Python version (& distribution if applicable, e.g. Anaconda): python 3.9, python 3.10

## Code Snippet

```python
from typing import Callable, Generic, TypeVar

from typing_extensions import ParamSpec

_Params = ParamSpec("_Params")
_Return = TypeVar("_Return")


class TestClass(Generic[_Params, _Return]):
    def __init__(self, func: Callable[_Params, _Return]) -> None:
        self._func = func

    def problematic_method(
        self,
        *args: _Params.args,
        test_param: int = 0,
        **kwargs: _Params.kwargs,
    ) -> _Return:
        """
        This method has additional parameter right after args for original function.
        """
        print(test_param)
        return self._func(*args, **kwargs)

    def not_a_problem(
        self,
        *args: _Params.args,
        **kwargs: _Params.kwargs,
    ) -> _Return:
        """Suggestions for this method are correct."""
        return self._func(*args, **kwargs)

    @classmethod
    def build(cls, func: Callable[_Params, _Return]) -> "TestClass[_Params, _Return]":
        return TestClass(func)


def test_func(a: int, b: int = 0) -> int:
    return a + b


inst = TestClass.build(test_func)

print(inst.not_a_problem(a=1, b=2))
print(inst.problematic_method(a=1, b=2, test_param=11))
```

## Actual behavior

This is what Pylance suggests: 

![image](https://user-images.githubusercontent.com/18153319/182042405-d77654ba-348e-426c-95a4-049853f82de0.png)

![image](https://user-images.githubusercontent.com/18153319/182042395-daf8c52e-26f6-40de-a68b-b5c1a569388c.png)

It also suggest not `a=, b=, test_param=`, but this: 

![image](https://user-images.githubusercontent.com/18153319/182043289-d22c412c-0db7-4acd-be2d-28d056ff66fe.png)

## Expected behavior

I expect pylance to add this additional keyword-only argument to the list of function's arguments. So the suggestion will look like this:

```
(a: int, b: int = ..., *, test_param: int = ...)
```

**Comment by s3rius** [comment]
Thanks for your speed.

**Comment by erictraut** [comment]
By default, pylance does not enable type checking diagnostics. If you enable this (by setting `python.analysis.typeCheckingMode` to "basic"), you will receive additional type-related errors and warnings. In this particular case, you will see an error that indicates why you're seeing the current behavior.

`Keyword parameter "test_param" cannot appear in signature after ParamSpec args parameter`

PEP 612 does not allow you to include keyword arguments between the `*args: P.args` and `**kwargs: P.kwargs` parameters in a signature that uses ParamSpecs. You can add positional parameters prior to the `*args` parameter. 

To find the relevant part of the PEP 612 spec, search for the following:

```python
  def bar(*args: P.args, s: str, **kwargs: P.kwargs) -> None:  # Rejected
    pass
```

So pyright is correct in flagging this as an invalid use of `ParamSpec`.

**Comment by s3rius** [comment]
Thanks for the quick reply.
